### PR TITLE
Build splunk-otel-collector Windows docker image for tests

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Resolves https://github.com/signalfx/splunk-otel-java/issues/137

Ideally we should use splunk-otel-collector everywhere, but that needs to wait until splunk-otel-collector has a Windows docker image.

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -32,5 +32,11 @@ jobs:
           distribution: adopt
           java-version: 11
 
+      - name: Build Windows Collector image
```
Comment (iNikem) [inline .github/workflows/pr.yaml:35]: Do we need to build this for every PR? Or we can just pull the image built by CI job?

**Review by iNikem — COMMENTED** [review]
Can we instead contribute to https://github.com/open-telemetry/opentelemetry-collector/issues/2715 ?

File: .github/workflows/pr.yaml (hunk 1 above)
Comment (mateuszrzeszutek) [reply .github/workflows/pr.yaml:35]: I'll change the CI job 👍

**Comment by mateuszrzeszutek** [comment]
> Can we instead contribute to open-telemetry/opentelemetry-collector#2715 ?

We're using the signalfx metrics receiver, so actually we'd like to get this one done: https://github.com/signalfx/splunk-otel-collector/issues/260

It'd be nice to have both of them, but I have no idea how go/collector build works 😞

File: smoke-tests/otel-collector-contrib-windows/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+FROM mcr.microsoft.com/windows/servercore:ltsc2019
+ADD https://github.com/signalfx/splunk-otel-collector/releases/latest/download/otelcol_windows_amd64.exe otelcol_windows_amd64.exe
```
Comment (iNikem) [inline smoke-tests/otel-collector-contrib-windows/Dockerfile:2]: I think we need contrib here, no?

Comment (mateuszrzeszutek) [reply smoke-tests/otel-collector-contrib-windows/Dockerfile:2]: 🤦  I forgot about the most important thing

**Review by iNikem — APPROVED** [review]

**Pull request merged**